// Copyright 2016 The Android Open Source Project
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto2";

package cuttlefish.acloud;

// This protobuffer is used to deserialize acloud configuration from a textproto
// file, it's never serialized to binary. The original file is
// https://cs.android.com/android/platform/superproject/main/+/main:tools/acloud/internal/proto/user_config.proto;drc=1bc201a83c70c204f21ad6a2c31d49cf1dd8ea4a
// Given that the original file has changed in many ways in the past, only the
// minimal set of fields that allow cvd to translate commands is included here.

// Hold configurations from user.
message UserConfig {
  // Compute Engine project name
  optional string project = 3;
  // Compute Engine zone name, e.g. "us-central1-f"
  optional string zone = 4;
  // [CVD only] Opt out from using cvdr.
  optional bool use_legacy_acloud = 39;
  optional string launch_args = 25;

  reserved "service_account_name", "service_account_private_key_path",
      "machine_type", "network", "ssh_private_key_path", "ssh_public_key_path",
      "storage_bucket_name", "orientation", "resolution", "client_id",
      "client_secret", "stable_host_image_name", "stable_host_image_family",
      "stable_host_image_project", "stable_goldfish_host_image_name",
      "stable_goldfish_host_image_project",
      "service_account_json_private_key_path", "hw_property",
      "stable_cheeps_host_image_name", "stable_cheeps_host_image_project",
      "instance_name_pattern", "extra_scopes", "extra_args_ssh_tunnel",
      "fetch_cvd_version", "betty_image", "api_key", "api_url", "oxygen_client",
      "oxygen_lease_args", "disk_type", "extra_data_disk_size_gb",
      "metadata_variable", "enable_multi_stage", "connect_hostname", "use_cvdr";
}
